relay-general/src/pii/processor.rs

```diff
@@ -195,6 +198,95 @@ impl<'a> Processor for PiiProcessor<'a> {
         replay.process_child_values(self, state)?;
         Ok(())
     }
+
+    fn process_event(
+        &mut self,
+        event: &mut Event,
+        _meta: &mut Meta,
+        state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        scrub_graphql(event);
+
+        event.process_child_values(self, state)?;
+
+        Ok(())
+    }
+}
+
+/// Scrubs GraphQL variables from the event.
+fn scrub_graphql(event: &mut Event) {
+    let mut keys: Option<BTreeSet<String>> = None;
```

Comment:
1. Please use `&str` instead of `String`.
2. Why wrap this in an `Option`?

```suggestion
    let mut keys: Option<BTreeSet<&str>> = None;
```

Reply:
2. if you init the `BTreeSet` every time `process_event` gets called, you pay the overhead of inviting a `BTreeSet` every time even if you don't need it.
The equivalent of a nullable variable in Rust is `Option` unless I'm missing something.

Reply:
https://doc.rust-lang.org/stable/std/collections/struct.BTreeSet.html#method.new

> Does not allocate anything on its own.

Removing the `Option`.